torch/distributed/_tensor/experimental/tensor_parallel.py

```diff
@@ -0,0 +1,426 @@
+import copy
+import operator
+from typing import Any, cast, Dict, List, Optional, Sequence, Set, Tuple
+
+import torch
+from torch.distributed._tensor import DeviceMesh, distribute_tensor, DTensor
+from torch.distributed._tensor.op_schema import (
+    DTensorSpec,
+    OpSchema,
+    OutputSharding,
+    OutputSpecType,
+    PlacementStrategy,
+)
+from torch.distributed._tensor.placement_types import (
+    Placement,
+    Replicate,
+    Shard,
+    TensorMeta,
+)
+from torch.distributed._tensor.redistribute import redistribute_local_tensor
+from torch.export import ExportedProgram
+from torch.export.exported_program import ExportGraphSignature
+from torch.fx import GraphModule
+from torch.fx.experimental.proxy_tensor import make_fx
+from torch.fx.node import Node
+from torch.fx.passes.infra.pass_base import PassBase, PassResult
+from torch.fx.passes.shape_prop import _extract_tensor_metadata
+
+
+aten = torch.ops.aten
+
+
+def tensor_parallel_transformation(
+    exported_program: ExportedProgram,
+    rank: int,
+    world_size: int,
+) -> ExportedProgram:
```

Comment:
Let's add a docstr to:
* explicitly mention that this is a graph transformation pass for torch.export to transform a single device graph to a tensor parallel graph
* add a warning here that this API is experimental and subject to change so that people don't abuse this API